x-pack/plugins/observability_solution/inventory/server/routes/entities/get_entity_groups.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ObservabilityElasticsearchClient } from '@kbn/observability-utils/es/client/create_observability_es_client';
+import { kqlQuery } from '@kbn/observability-utils/es/queries/kql_query';
+import { esqlResultToPlainObjects } from '@kbn/observability-utils/es/utils/esql_result_to_plain_objects';
+import { ENTITY_TYPE } from '@kbn/observability-shared-plugin/common';
+import { ScalarValue } from '@elastic/elasticsearch/lib/api/types';
+import {
+  ENTITIES_LATEST_ALIAS,
+  type EntityGroup,
+  MAX_NUMBER_OF_ENTITIES,
+} from '../../../common/entities';
+import { getBuiltinEntityDefinitionIdESQLWhereClause } from './query_helper';
+
+export async function getEntityGroupsBy({
+  inventoryEsClient,
+  field,
+  kuery,
+  entityTypes,
+}: {
+  inventoryEsClient: ObservabilityElasticsearchClient;
+  field: string;
+  kuery?: string;
+  entityTypes?: string[];
+}) {
+  const from = `FROM ${ENTITIES_LATEST_ALIAS}`;
+  const where: string[] = [getBuiltinEntityDefinitionIdESQLWhereClause()];
```

Comment:
Do you need to manually type it?
```suggestion
  const where = [getBuiltinEntityDefinitionIdESQLWhereClause()];
```